testinfra/modules/interface.py

```diff
@@ -74,6 +128,37 @@ def addresses(self):
                 addrs.append(splitted[1].split("/", 1)[0])
         return addrs
 
+    @property
+    def is_up(self):
+
+        # Check Connection Carrier Status
+        if int(self.check_output(
+                "cat /sys/class/net/%s/carrier", self.name)) == 1:
+
+            # Check Operator Status
+            if str(self.check_output(
+                    "cat /sys/class/net/%s/operstate", self.name)) == "up":
+                return True
+        else:
+            return False
+
+    @property
+    def mtu(self):
+        return int(self.check_output(
+            "cat /sys/class/net/%s/mtu", self.name))
+
+    @property
+    def is_bond(self):
+        return self.run_test("ip link show %s type bond", self.name).rc == 0
```

Comment:
(Sorry for delay) I think the command is wrong:

```
$ ip link show eth0 type bond
Error: either "dev" is duplicate, or "type" is a garbage.
$ echo $?
255
```

Ideally this deserve tests, for example in test/test_modules.py:

```python
def test_interface(host):
    iface = host.interface("eth0")
    assert iface.is_up
    assert iface.mtu > 0
    assert iface.speed > 0
    assert not iface.is_bond
    assert not iface.is_bridge
    assert not iface.is_vlan
```